bgpd/bgp_nht.c

```diff
@@ -206,9 +228,16 @@ int bgp_find_or_add_nexthop(struct bgp *bgp_route, struct bgp *bgp_nexthop,
 		 * Gather the ifindex for if up/down events to be
 		 * tagged into this fun
 		 */
-		if (afi == AFI_IP6
-		    && IN6_IS_ADDR_LINKLOCAL(&peer->su.sin6.sin6_addr))
+		if (afi == AFI_IP6 &&
+		    IN6_IS_ADDR_LINKLOCAL(&peer->su.sin6.sin6_addr)) {
 			ifindex = peer->su.sin6.sin6_scope_id;
+			if (ifindex == 0 && BGP_DEBUG(nht, NHT)) {
+				zlog_debug(
+					"%s: Unable to locate ifindex, waiting till we have one",
+					peer->conf_if);
+				return 0;
```

Comment:
Will we return only if debug is enabled?

Reply:
fuxed